server/swap/swap.go

```diff
@@ -203,10 +207,23 @@ func NewSwapper(cfg *Config) *Swapper {
 	authMgr.RegisterMethod(msgjson.InitRoute, swapper.handleInit)
 	authMgr.RegisterMethod(msgjson.RedeemRoute, swapper.handleRedeem)
 
+	swapper.wg.Add(1)
 	go swapper.start()
 	return swapper
 }
 
+// Stop begins the Swapper's shutdown. Use WaitForShutdown after Stop to wait
+// for shutdown to complete.
+func (s *Swapper) Stop() {
+	log.Infof("Swapper shutting down...")
+	s.cancel()
+}
+
+// WaitForShutdown waits until the main swapper loop is finished.
+func (s *Swapper) WaitForShutdown() {
+	s.wg.Wait()
+}
+
```

Comment:
I'm not at all certain that this is sufficient for Swapper, but a way to allow the goroutines go return gracefully is needed.